app/services/super_memo.rb

```diff
@@ -0,0 +1,55 @@
+class SuperMemo
+  GOOD_DELAY = 1
+  NORMAL_DELAY = 3
+
+  def initialize(params)
+    @time = Time.at(params[:time].to_f/1000).strftime("%M").to_f
+    @complexity = params[:complexity] ? (params[:complexity] < 1.3 ? 1.3 : params[:complexity]) : 2.5
+    @times_reviewed = params[:times_reviewed]
+    @typos = params[:typos]
+    @word_length = params[:word_length]
+  end
+
+  def response_quality
+    if @time <= GOOD_DELAY
+      @good_delay = true
+    elsif @time > GOOD_DELAY and @time < NORMAL_DELAY
+      @normal_delay = true
+    else
+      @bad_delay = true
+    end
+  end
+
+  def rating
+    response_quality
+
+    if @good_delay and @typos.zero?
+      5
+    elsif @normal_delay and @typos.zero?
+      4
+    elsif @bad_delay and @typos.zero?
+      3
+    elsif @typos <= 3 and @typos < @word_length
+      2
+    elsif @typos >= 4 and @typos < @word_length
+      1
+    elsif @typos >= @word_length
+      0
+    end
+  end
+
+  def run
+    case @times_reviewed
+    when 0
+      1
+    when 1
+      3
+    else
+      q = rating
+      ( (@times_reviewed - 1) * (@complexity + (0.1-(5-q)*(0.08+(5-q)*0.02))) ).round
+    end
+  end
+
+private
```

Comment:
Indent access modifiers like `private`.<br>Keep a blank line before and after `private`.